samples/socket-api/wasm-src/timeout_server.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+#include <unistd.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#endif
+
+int
+main(int argc, char *argv[])
+{
+    int socket_fd;
+    int client_socket_fd;
+    struct sockaddr_in addr = { 0 };
+    int addrlen = sizeof(addr);
+    int bool_opt = 1;
+
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(1234);
+    addr.sin_addr.s_addr = htonl(INADDR_ANY);
+    if ((socket_fd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        perror("Create socket failed");
+        return EXIT_FAILURE;
+    }
+
+    if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &bool_opt,
+                   sizeof(bool_opt))
+        == -1) {
+        perror("Failed setting SO_REUSEADDR");
+        close(socket_fd);
+        return EXIT_FAILURE;
+    }
+
+    if (bind(socket_fd, (struct sockaddr *)&addr, addrlen) == -1) {
+        perror("Bind socket failed");
+        close(socket_fd);
+        return EXIT_FAILURE;
+    }
+
+    if (listen(socket_fd, 1) == -1) {
+        perror("Listen failed");
+        close(socket_fd);
+        return EXIT_FAILURE;
+    }
+
+    if ((client_socket_fd =
+             accept(socket_fd, (struct sockaddr *)&addr, (socklen_t *)&addrlen))
+        == -1) {
+        perror("Accept failed");
+        close(socket_fd);
+        return EXIT_FAILURE;
+    }
+
+    printf("Client connected, sleeping for 10s\n");
+    sleep(10);
+
+    printf("Shuting down\n");
+    shutdown(client_socket_fd, SHUT_RDWR);
```

Comment:
Should also `close(client_socket_fd)`, the static analysis tool Coverity reports that not all resources are destroyed if close isn't called.

Reply:
+1 does WAMR have a Coverity project setup / is it supposed to run as part of the github checks? Not sure why, but this commit didn't trigger any checks.

Reply:
I'm happy to run this manually before publishing a PR. @wenyongh Maybe it's something that can be included as part of a contributor guide readme?

Reply:
@cimacmillan The Coverity server is setup internally, we need to apply the privileges to access it, don't known how to make it available for the Github developers or CI to run, we will investigate more.